### ISSUE TITLE
Haul creating random folders in each run in project root

### ISSUE THREAD

**Issue by sirajulm** [callstack/haul#519]
### Current Behavior

Haul is pacakging the bundle and is being successfully loaded in both iOS and Android. There is no issue with the build, but haul is creating random folders in project `root`, prefixed as `haul-start-` while running Haul via `yarn haul start` command. 

Please refer the screenshot:
![image](https://user-images.githubusercontent.com/2260996/52916391-d2247080-3304-11e9-87bb-63e48f5e61c3.png)

 
### Expected Behavior

Haul should not be creating random folders each time it is being called, which would pollute the project root. If those folders are unavoidable either create single folder each time the command is run, which can then be added to `.gitignore` or provide option to the user to provide a path like `/dist` or `/build` to which these folders can be diverted.

### Haul Configuration (webpack.haul.js)

```
import { createWebpackConfig } from "haul";
import path from "path";

export default {
  webpack: (env, defaults) => {
    const config = createWebpackConfig({
      // ...defaults,
      entry: `./index.js`,
      resolve: {
        // ...defaults.resolve,
        alias: {
          // ...defaults.resolve.alias,
          react: path.join(__dirname, "node_modules/react"),
          "react-native": path.join(__dirname, "node_modules/react-native")
        }
      }
    })(env);

    config.module.rules = [
      {
        test: /\.tsx?$/,
        loader: "ts-loader"
      },
      ...config.module.rules
    ];

    config.resolve.extensions = [
      ".js",
      ".ts",
      ".tsx",
      `.${env.platform}.ts`,
      ".native.ts",
      `.${env.platform}.tsx`,
      ".native.tsx",
      ...config.resolve.extensions
    ];

    return config;
  }
};

```

### Your Environment

| software         | version
| ---------------- | -------
| Haul             |  ^1.0.0-rc.13
| react-native     |  0.58.4
| node             | v11.6.0
| npm or yarn      | 1.13.0

**Comment by sirajulm** [comment]
After a lot of digging found the solution, needed to add `--assetDest` option to specify a destination. This is found missing the the documentation.

Related to Issue #513 which is fixed in PR #516 

Closing the issue.

**Comment by thymikee** [comment]
Btw by default the asset destination is set to os tmp directory

**Comment by sirajulm** [comment]
Yeah, have checked the code, and it points to tmp directory. But in Mac I'm getting the folders being created in the project root. Not sure why this happens. :-(

**Comment by sirajulm** [comment]
@thymikee isn't the temp directory be created like
`assetsDest = fs.mkdtempSync(path.join(os.tmpdir(), "haul-start-"))`  instead of 
`assetsDest = path.join(os.tmpdir(), fs.mkdtempSync('haul-start-'))`

**Comment by thymikee** [comment]
Oh, yea. Is it like that really? I'd gladly appreciate a fix

**Comment by sirajulm** [comment]
Yes, otherwise, I'm getting the folder being created in `cwd`, but assetDest gives the path of `tmp` which is non-existent. Will create a PR in a while

**Comment by daryl** [comment]
Not sure why this has been closed this happens to me currently.

**Comment by sirajulm** [comment]
@daryl This has been closed because we did fix the issue and have released the changes. 
We could reopen it if you could provide us the version of haul and steps to reproduce. Thanks.

**Comment by daryl** [comment]
Wouldn't say it's fixed? I've tried both `@latest` and `@next`. Always get a folder.

**Comment by sirajulm** [comment]
Hey daryl, will look into it and get back to you :)

**Comment by sirajulm** [comment]
@daryl  Unfortunately, I think the changes are not pushed to npm repository, which still points to version `1.0.0-rc.13`.

This have been fixed from `1.0.0-rc.14` and `next`. I've verified it now.
You could instead try installing 
`yarn add -D https://github.com/callstack/haul#v1.0.0-rc.14`
 or
`yarn add -D https://github.com/callstack/haul#next`

Will reopen the ticket until it will be released to npm.

Thanks for pointing out.

**Comment by thymikee** [comment]
As confirmed with @sirajulm this is published under `next` tag (it points to `1.0.0-rc.15` now).
